### PR TITLE
Capture exceptions during test teardown

### LINKED ISSUE #65: Handling of exception during teardown.
When doing the test teardown, pytest-qt does this:

```python
@pytest.mark.hookwrapper
def pytest_runtest_teardown():
    """
    Hook called after each test tear down, to process any pending events and
    avoiding leaking events to the next test.
    """
    yield
    app = QApplication.instance()
    if app is not None:
        app.processEvents()
```

PyQt 5.5 [changed its behaviour](http://pyqt.sourceforge.net/Docs/PyQt5/incompatibilities.html#pyqt-v5-5) to call `abort()` if there is an exception in a Qt virtual method and no `sys.excepthook` is set.

This means if an exception happens during that part, instead of it being silently swallowed (as it's not handled by pytest-qt anymore), in my testfault I got a failure (without much info if running via tox):

```
$  tox -e unittests -- tests/misc/test_guiprocess.py
[...]
tests/misc/test_guiprocess.py ......ERROR: InvocationError: '/home/florian/proj/qutebrowser/git/.tox/unittests/bin/python -m py.test --strict -rfEsw tests/misc/test_guiprocess.py'
```

Fortunately using `pytest-faulthandler`, `-s` and `-v` reveals a bit more:

```
$ tox -e unittests -- -v -s tests/misc/test_guiprocess.py
[...]
tests/misc/test_guiprocess.py::test_cmd_args PASSEDTraceback (most recent call last):
  File "/home/florian/proj/qutebrowser/git/qutebrowser/utils/objreg.py", line 188, in _get_window_registry
    return win.registry
AttributeError: 'NoneType' object has no attribute 'registry'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/florian/proj/qutebrowser/git/qutebrowser/misc/guiprocess.py", line 94, in on_error
    self._what, msg), immediately=True)
  File "/home/florian/proj/qutebrowser/git/qutebrowser/utils/message.py", line 121, in error
    _wrapper(win_id, 'error', message, immediately)
  File "/home/florian/proj/qutebrowser/git/qutebrowser/utils/message.py", line 52, in _wrapper
    bridge = _get_bridge(win_id)
  File "/home/florian/proj/qutebrowser/git/qutebrowser/utils/message.py", line 86, in _get_bridge
    return objreg.get('message-bridge', scope='window', window=win_id)
  File "/home/florian/proj/qutebrowser/git/qutebrowser/utils/objreg.py", line 215, in get
    reg = _get_registry(scope, window, tab)
  File "/home/florian/proj/qutebrowser/git/qutebrowser/utils/objreg.py", line 204, in _get_registry
    return _get_window_registry(window)
  File "/home/florian/proj/qutebrowser/git/qutebrowser/utils/objreg.py", line 190, in _get_window_registry
    raise RegistryUnavailableError('window')
qutebrowser.utils.objreg.RegistryUnavailableError: window
Fatal Python error: Aborted

Current thread 0x00007f8e31e78700 (most recent call first):
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/pytestqt/plugin.py", line 592 in pytest_runtest_teardown
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/_pytest/core.py", line 109 in wrapped_call
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/_pytest/core.py", line 393 in execute
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/_pytest/core.py", line 123 in __init__
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/_pytest/core.py", line 107 in wrapped_call
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/_pytest/core.py", line 393 in execute
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/_pytest/core.py", line 528 in _docall
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/_pytest/core.py", line 521 in __call__
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/_pytest/runner.py", line 137 in <lambda>
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/_pytest/runner.py", line 149 in __init__
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/_pytest/runner.py", line 137 in call_runtest_hook
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/_pytest/runner.py", line 119 in call_and_report
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/_pytest/runner.py", line 77 in runtestprotocol
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/_pytest/runner.py", line 65 in pytest_runtest_protocol
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/_pytest/core.py", line 394 in execute
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/_pytest/core.py", line 123 in __init__
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/_pytest/core.py", line 107 in wrapped_call
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/_pytest/core.py", line 393 in execute
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/_pytest/core.py", line 528 in _docall
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/_pytest/core.py", line 521 in __call__
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/_pytest/main.py", line 142 in pytest_runtestloop
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/_pytest/core.py", line 394 in execute
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/_pytest/core.py", line 528 in _docall
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/_pytest/core.py", line 521 in __call__
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/_pytest/main.py", line 122 in _main
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/_pytest/main.py", line 84 in wrap_session
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/_pytest/main.py", line 116 in pytest_cmdline_main
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/_pytest/core.py", line 394 in execute
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/_pytest/core.py", line 528 in _docall
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/_pytest/core.py", line 521 in __call__
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/_pytest/config.py", line 41 in main
  File "/home/florian/proj/qutebrowser/git/.tox/unittests/lib/python3.4/site-packages/py/test.py", line 4 in <module>
  File "/usr/lib64/python3.4/runpy.py", line 85 in _run_code
  File "/usr/lib64/python3.4/runpy.py", line 170 in _run_module_as_main
```

I can't reproduce it in a minimal example, but you can clone qutebrowser and run the commandlines above.

I'll of course now fix the exception, but pytest-qt should have some way to handle them (perhaps by using `capture_exceptions()` in `pytest_runtest_teardown` as well?).

### PR DESCRIPTION
**Pull request by nicoddemus**
Fixes #65

### CODE REVIEW

File: pytestqt/_tests/test_exceptions.py [hunk 1]
```diff
@@ -58,6 +58,8 @@ def test_format_captured_exceptions():
 
 
 @pytest.mark.parametrize('no_capture_by_marker', [True, False])
+@pytest.mark.skipif(QT_API == 'pyqt5', reason='non captured exceptions on PyQt'
+                                              ' 5.5+ crash the interpreter.')
```
Comment (The-Compiler) [inline pytestqt/_tests/test_exceptions.py]: You could probably still run this by setting `sys.excepthook` to some function which just ignores it - then `qFatal`/`abort` won't be called.

**Comment by The-Compiler** [comment]
Thanks for the quick fix, Bruno! :smile:

I can confirm it works as expected.

File: pytestqt/_tests/test_exceptions.py (hunk 1 above)
Comment (nicoddemus) [inline pytestqt/_tests/test_exceptions.py]: Good idea, thanks! :smile:

**Comment by ctismer** [comment]
Hi guys,
this is a bit off-topic, but I wanted to let you know that I'm quite far with PySide2 for Qt5.
If you are interested, then I'd be very happy to get your input, suggestions or even help.

Right now, 50% of tests are crashing (or, to be positive, already half of the tests pass).

I am now in the last phase: hunting all the bugs and hoping that this converges in the end.
And I have not supported signals, yet, and I need input on that.

You find everything on Github, in the official PySide account. My versions end in "...2".

best - Chris

**Comment by nicoddemus** [comment]
Hi @ctismer! I wasn't even aware there was a PySide2 in the works, thanks for the work.

I will take a look at the project and see if I have any suggestions and perhaps some help.

Cheers :beers:

**Comment by ctismer** [comment]
Hi @nicoddemus ! Yes, this stuff is online since 2 weeks, and nobody expected it.
I am working on that since end of May, and I'm exhausted, what should I say.
Would be great to talk to people who are actually solving problems ;-)

best - Chris

**Comment by The-Compiler** [comment]
@ctismer Thanks for your work! I've been following it a bit on the PySide mailinglist(s).

Unfortunately I won't really be able to help as I'm already swamped with my own project and other contributions - but I'm looking forward to testing it with my project ([http://www,qutebrowser.org/](qutebrowser)) once it's in some usable state.

**Comment by ctismer** [comment]
that's fine with me. I just wanted to make sure that you know that PySide is growing Qt5 support.

**Pull request merged**